test/unit/org/apache/cassandra/service/DiskFailurePolicyTest.java

```diff
@@ -124,12 +135,29 @@ public void teardown()
     public void testPolicies()
     {
         DatabaseDescriptor.setDiskFailurePolicy(testPolicy);
-        JVMStabilityInspector.inspectThrowable(t);
-        Assert.assertEquals(expectJVMKilled, killerForTests.wasKilled());
-        Assert.assertEquals(expectJVMKilledQuiet, killerForTests.wasKilledQuietly());
-        if (!expectJVMKilled) {
+        try
+        {
+            JVMStabilityInspector.inspectThrowable(t);
+        }
+        catch (OutOfMemoryError e)
```

Comment:
this is eventually thrown by `JVMStabilityInspector.inspectThrowable(t);` if you follow it deeply. We just need to make the difference between oom thrown as part of the test and oom thrown _while running a test_.